spark-operator-api/src/main/java/org/apache/spark/k8s/operator/spec/ResourceRetentionPolicy.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.spec;
+
+public enum ResourceRetentionPolicy {
```

Comment:
We need to rename the values of `ResourceRetentionPolicy` in order to be consistent with K8s standard.

For example, the following is K8s's naming convention and also this PR follows in `restartPolicy`.
- https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#example
```yaml
restartPolicy: OnFailure
```

Given the above, we need the following style instead of `RetainOnFailure`.
```yaml
retainPolicy: OnFailure
```